docs/style_guide/definitions.md

```diff
@@ -0,0 +1,103 @@
+# Definitions (Top level definitions)
```

Comment:
Definitions seems redundant shall we use only Top Level Deffinitions

Reply:
Will change the title